docs: rewrite README (#344)
withastro/prettier-plugin-astro: README.md

# [Prettier](https://prettier.io/) Plugin for [Astro](https://astro.build/)

Official Prettier plugin adding support for formatting `.astro` files.

## Installation

```shell
npm i --save-dev prettier-plugin-astro prettier
```

### Recommended configuration

> For optimal compatibility with the different package managers and Prettier plugins, we recommend manually specifying your plugins and the parser to use for Astro files in your Prettier config, like such:

Reviewer: ```suggestion
For optimal compatibility with the different package managers and Prettier plugins, we recommend manually specifying your plugins and the parser to use for Astro files in your Prettier config as shown in the example below:
```

```js
module.exports = {
  plugins: [require.resolve('prettier-plugin-astro')],
  overrides: [
    {
      files: '*.astro',
      options: {
        parser: 'astro',
      },
    },
  ],
};
```

> The `require.resolve` call can alternatively be changed to a direct path, like such: `plugins: ["./node_modules/prettier-plugin-astro"]` for usage inside a non-JS config file.

Reviewer: ```suggestion
Alternatively, for use inside a non-JS config file, the `require.resolve` call can be changed to a direct path, such as `plugins: ["./node_modules/prettier-plugin-astro"]`.
```
Just front-loading the why they might need to know this part, highlighting that it's an alternative (not something extra they need to do/know) and when to consider this alternative.

To customize formatting behavior, see the [Configuration](#configuration) section below.

## Using in VS Code

> **Note**
> If you're using [Astro's VS Code extension](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode), Prettier and this plugin are already included. Only follow the guide below to format using Prettier's official extension.

> First install the [VS Code Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and add the following settings to your VS Code configuration so VS Code is aware that Prettier can be used for Astro files and set it as the default formatter for Astro files:

Reviewer: ```suggestion
First install the [VS Code Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and add the following settings to your VS Code configuration:
```

```json
{
  "prettier.documentSelectors": ["**/*.astro"],
  "[astro]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  }
}
```

> When submitting issues about formatting in VS Code, please specify if you're using the Astro extension or the Prettier extension to format your files.

Reviewer: ```suggestion
 The settings above ensure that VS Code is aware that Prettier can be used for Astro files, and sets Prettier as the default formatter for Astro files.

### Reporting issues

When submitting issues about formatting your `.astro` files in VS Code, please specify which extension you are using to format your files: Astro's own extension or the Prettier extension.
```
2 things:
- I put the explanation of the settings underneath, so that the instructions to follow were a simple one-liner (and, people can skip WHY this works and what it's doing if they want)
- I made a separate heading for reporting issues because people have been told to skip this section if they're using the Astro extension, so they might not even see this. I made this a L3 heading under the VS code section, since it is related to VS Code. But, if you felt like you wanted to call more attention to this with an L2 heading, I think that would be fine too.
